core/unit_test/TestReduceCombinatorical.hpp

```diff
@@ -394,58 +394,69 @@ struct TestReduceCombinatoricalInstantiation {
   }
 
   template <class... Args>
-  static void AddReturnArgument(Args... args) {
-    Kokkos::View<double, Kokkos::HostSpace> result_view("ResultView");
-    double expected_result = 1000.0 * 999.0 / 2.0;
+  static void AddReturnArgument(int N, Args... args) {
+    Kokkos::View<double, Kokkos::HostSpace> result_view("ResultViewHost");
+    Kokkos::View<double, ExecSpace> result_view_device("ResultViewDevice");
+    double expected_result = (1.0 * N) * (1.0 * N - 1.0) / 2.0;
 
-    double value = 0;
+    double value = 99;
     Kokkos::parallel_reduce(args..., value);
     ASSERT_EQ(expected_result, value);
 
-    result_view() = 0;
+    result_view() = 99;
     CallParallelReduce(args..., result_view);
     Kokkos::fence();
     ASSERT_EQ(expected_result, result_view());
 
-    value = 0;
+#ifndef KOKKOS_ENABLE_OPENMPTARGET
+    result_view() = 99;
+    CallParallelReduce(args..., result_view_device);
+    Kokkos::fence();
+    Kokkos::deep_copy(result_view, result_view_device);
+    ASSERT_EQ(expected_result, result_view());
+#endif
+
+    value = 99;
     CallParallelReduce(
         args...,
         Kokkos::View<double, Kokkos::HostSpace,
                      Kokkos::MemoryTraits<Kokkos::Unmanaged> >(&value));
     Kokkos::fence();
     ASSERT_EQ(expected_result, value);
 
-    result_view() = 0;
+    result_view() = 99;
     const Kokkos::View<double, Kokkos::HostSpace,
                        Kokkos::MemoryTraits<Kokkos::Unmanaged> >
         result_view_const_um = result_view;
     CallParallelReduce(args..., result_view_const_um);
     Kokkos::fence();
     ASSERT_EQ(expected_result, result_view_const_um());
 
-    value = 0;
+    value = 99;
 // WORKAROUND OPENMPTARGET Custom Reducers not implemented
 #ifndef KOKKOS_ENABLE_OPENMPTARGET
     CallParallelReduce(args...,
                        Test::ReduceCombinatorical::AddPlus<double>(value));
     if ((Kokkos::DefaultExecutionSpace::concurrency() > 1) &&
-        (ExecSpace::concurrency() > 1)) {
+        (ExecSpace::concurrency() > 1) && (expected_result > 0)) {
```

Comment:
Why do we need to guard the `N=0` case here?

Reply:
because for N==0 it would result in value==expected_result, while for every other case with N>0 (due to the way the team functor works) value>expected_result.

Reply:
Yeah, it seems I forgot to delete this comment.